File: MB.Data.Entities/Transaction.cs
using System;

namespace MB.Data.Entities
{
    public class Transaction
    {
        public int ID { get; set; }
        public string Description { get; set; }
        public decimal Amount { get; set; }
        public DateTime DateTime { get; set; }

        public decimal OriginalAmount { get; set; }
        public int CurrencyId { get; set; }
        public virtual Currency Currency { get; set; }

        public int AccountId { get; set; }
        public virtual Account Account { get; set; }
    }
}

File: MBApplicationUI/src/app/model/transaction.ts
import { Currency } from "./currency";

export class Transaction {
  public id?: number;
  public description: string;
  public amount?: number;
  public originalAmount: number;
  public currencyId: number;
  public currency?: Currency;
  public accountId: number;
  public dateTime?: string;

  constructor(description: string, originalAmount: number, currencyId: number, accountId: number){
    this.description = description;
    this.originalAmount = originalAmount;
    this.currencyId = currencyId;
    this.accountId = accountId;
  }
}
